content/exchange/artifacts/SuspiciousWMIConsumers

```diff
@@ -0,0 +1,148 @@
+name: Windows.Analysis.SuspiciousWMIConsumers
+
+description: |
+  This artifact reports suspicious WMI Event Consumers and their associated Filters
+  that may indicate a malicious abuse for persistence.
+
+  NOTE: This artifact uses the same logic as Windows.Persistence.PermanentWMIEvents 
```

Comment:
I would consider just calling the underlying artefact like:
SELECT * FROM Artifact.Windows.Persistence.PermanentWMIEvents(..)
instead of copy pasting all the VQL. Then your just looking at maintaining the lookups.

Reply:
@mgreen27 at the beginning, I was calling the original artifact then I thought I might need to copy the VQL because I've injected some WHERE clauses to enhance the performance by limiting the number of rows that I'm looping on to apply the regex, please let me know if it doesn't matter and I'll call the original artifact again instead.

Reply:
Im really impressed with that thinking!

When you list all I dont think WMI matters too much from a performance viewpoint but I notice your adding into the consumer section so that will give some improvement.  Maybe its worth adding a filter into the main project too?

Reply:
Thanks @mgreen27! sure, just let me know if you need to add this artifact as an analysis artifact to the main project, or do you need to just add the WHERE clause to the original artifact (which I don't recommend) and I'll do it!